app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -236,6 +240,13 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
       return string.substring(0, stringToFind.length) === stringToFind
     },
 
+    stringEndsWith: function (string, stringToFind) {
+      if (stringToFind.length > string.length) {
```

Comment:
I think you're right that I should change it - I was worried that if `stringToFind`  was longer than the `string` it would cause a problem as the start index for the substring would be negative, but it seems that JS allows a negative index.